addons/purchase/models/purchase.py

```diff
@@ -1003,6 +1008,10 @@ def _compute_analytic_id_and_tag_ids(self):
             rec.account_analytic_id = rec.account_analytic_id or default_analytic_account.analytic_id
             rec.analytic_tag_ids = rec.analytic_tag_ids or default_analytic_account.analytic_tag_ids
 
+    @api.onchange('currency_id')
+    def onchange_currency_id(self):
```

Comment:
Onchange should be always private (begin with `_`).

Instead of create a new onchange which call a other one (`_onchange_quantity`) why  not juste add currency_id in decorator of `_onchange_quantity`:
```python
    @api.onchange('product_qty', 'product_uom')
    def _onchange_quantity(self):
```